### PR TITLE
npu fused op

### PR DESCRIPTION
**Pull request by ll819214**
## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/layers/rotary_embedding.py [hunk 1]
```diff
@@ -152,6 +161,27 @@ def forward_native(
...
+    def forward_npu(
+        self,
+        positions: torch.Tensor,
+        query: torch.Tensor,
+        key: torch.Tensor,
+        offsets: Optional[torch.Tensor] = None,
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        """A PyTorch-npu implementation of forward()."""
+        import os
+        if os.environ["SGLANG_ENABLE_TORCH_COMPILE"] == "1":
```
Comment (Alcanderian) [inline python/sglang/srt/layers/rotary_embedding.py]: use get_bool_env_var from utils

File: python/sglang/srt/layers/activation.py [hunk 2]
```diff
@@ -48,6 +48,13 @@
 
 logger = logging.getLogger(__name__)
 
+if is_npu():
+    try:
+        import torch_npu
+    except ImportError:
+        logger.warning("torch_npu is not installed. NPU support will be disabled.")
+        _is_npu = False
```
Comment (Alcanderian) [inline python/sglang/srt/layers/activation.py]: unused

File: python/sglang/srt/layers/activation.py [hunk 3]
```diff
@@ -48,6 +48,13 @@
 
 logger = logging.getLogger(__name__)
 
+if is_npu():
+    try:
+        import torch_npu
+    except ImportError:
```
Comment (Alcanderian) [inline python/sglang/srt/layers/activation.py]: Is logic in `is_npu`(https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/utils.py#L159) ensure the presence of torch_npu?

**Review by Alcanderian — APPROVED** [review]
